public/client-v2/src/components/FrontsEdit/Fronts.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import filterFrontsByPriority from '../../util/filterFrontsByPriority';
+import getFrontsConfig from '../../actions/FrontsConfig';
+import type { FrontDetail, FrontsConfig } from '../../types/Fronts';
+import type { State } from '../../types/State';
+
+type FrontsComponentProps = {
+  match: Object,
+  frontsActions: Object,
+  frontsConfig: FrontsConfig
+};
+
+type ComponentState = {
+  selectedFront: ?string
+};
+
+class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
+  state = {
+    selectedFront: null
+  };
+
+  componentDidMount() {
+    this.props.frontsActions.getFrontsConfig();
+  }
+
+  renderSelectPrompt = () => {
+    if (!this.state.selectedFront) {
+      return <option value="">Choose a front...</option>;
+    }
+    return null;
+  };
+
+  renderFrontOption = (front: FrontDetail) => (
+    <option value={front.id} key={front.id}>
+      {front.id}
+    </option>
+  );
+
+  render() {
+    if (!this.props.frontsConfig.fronts) {
+      return <div>Loading</div>;
+    }
+
+    const { match: { params: { priority } } } = this.props;
+    const filteredFronts: Array<FrontDetail> = filterFrontsByPriority(
```

Comment:
Here is a really good case for using `reselect` so that we can memoize this data when calling `mapStateToProps` so that other components can take advantage of it. I can come over and show you what I mean as I'm sure this makes no sense!